### PR TITLE
[ErrorCatcher] Make IDEs and static analysis tools happy

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | yes 
| Tests pass?   | yes    
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/Symfony/Component/ErrorHandler/ErrorRenderer/ErrorRenderer.php [hunk 1]
```diff
@@ -62,7 +62,7 @@ public function render($exception, string $format = 'html'): string
             throw new ErrorRendererNotFoundException(sprintf('No error renderer found for format "%s".', $format));
         }
 
-        if (!$exception instanceof FlattenException) {
+        if ($exception instanceof \Exception) {
```
Comment (stof) [inline src/Symfony/Component/ErrorHandler/ErrorRenderer/ErrorRenderer.php]: Shouldn't we check for Traversable here ? I think we need to let the component render `Error` too

Comment (fabpot) [inline src/Symfony/Component/ErrorHandler/ErrorRenderer/ErrorRenderer.php]: Also makes mere mortal like me reason about the code more easily as `create()` takes an `\Exception` instance.

Comment (fabpot) [reply src/Symfony/Component/ErrorHandler/ErrorRenderer/ErrorRenderer.php]: You mean `Throwable`?

Comment (stof) [reply src/Symfony/Component/ErrorHandler/ErrorRenderer/ErrorRenderer.php]: yeah, indeed.

Comment (fabpot) [reply src/Symfony/Component/ErrorHandler/ErrorRenderer/ErrorRenderer.php]: Done.

File: src/Symfony/Component/ErrorHandler/ErrorRenderer/ErrorRenderer.php [hunk 2]
```diff
@@ -62,8 +62,8 @@ public function render($exception, string $format = 'html'): string
             throw new ErrorRendererNotFoundException(sprintf('No error renderer found for format "%s".', $format));
         }
 
-        if (!$exception instanceof FlattenException) {
-            $exception = FlattenException::create($exception);
+        if ($exception instanceof \Throwable) {
```
Comment (stof) [inline src/Symfony/Component/ErrorHandler/ErrorRenderer/ErrorRenderer.php]: the signature of these methods in this new component should be changed to `\Throwable|FlattenException`

File: src/Symfony/Component/ErrorHandler/ExceptionHandler.php [hunk 3]
```diff
@@ -160,8 +160,8 @@ public function handle(\Exception $exception)
      */
     public function sendPhpResponse($exception)
```
Comment (yceruto) [inline src/Symfony/Component/ErrorHandler/ExceptionHandler.php]: https://github.com/symfony/symfony/pull/32221/files#r298199467 same here

File: src/Symfony/Component/ErrorHandler/ErrorRenderer/ErrorRenderer.php (hunk 2 above)
Comment (fabpot) [reply src/Symfony/Component/ErrorHandler/ErrorRenderer/ErrorRenderer.php]: indeed, done

File: src/Symfony/Component/ErrorHandler/ExceptionHandler.php (hunk 3 above)
Comment (fabpot) [reply src/Symfony/Component/ErrorHandler/ExceptionHandler.php]: done

**Pull request merged**